compiler/rustc_llvm/llvm-wrapper/CoverageMappingWrapper.cpp

```diff
@@ -62,11 +62,32 @@ fromRust(LLVMRustCounterMappingRegionKind Kind) {
   report_fatal_error("Bad LLVMRustCounterMappingRegionKind!");
 }
 
+#if LLVM_VERSION_GE(18, 0)
+// FFI equivalent of struct `llvm::coverage::CounterMappingRegion::MCDCParameters`
+// https://github.com/llvm/llvm-project/blob/ac406b481794efbc66122495d0973ae91c1bb046/llvm/include/llvm/ProfileData/Coverage/CoverageMapping.h#L253
+struct LLVMRustMCDCParameters {
+  uint32_t BitmapIdx;
+  uint32_t NumConditions;
+  uint32_t ID;
+  uint32_t TrueID;
+  uint32_t FalseID;
+};
+
+static coverage::CounterMappingRegion::MCDCParameters fromRust(
+    LLVMRustMCDCParameters Params) {
+  return {Params.BitmapIdx, Params.NumConditions, Params.ID, Params.TrueID,
+          Params.FalseID};
+}
+#endif
+
 // FFI equivalent of struct `llvm::coverage::CounterMappingRegion`
 // https://github.com/rust-lang/llvm-project/blob/ea6fa9c2/llvm/include/llvm/ProfileData/Coverage/CoverageMapping.h#L211-L304
 struct LLVMRustCounterMappingRegion {
   LLVMRustCounter Count;
   LLVMRustCounter FalseCount;
+#if LLVM_VERSION_GE(18, 0)
+  LLVMRustMCDCParameters MCDCParams;
+#endif
```

Comment:
It looks like you adjusted the C-side struct without adjusting the Rust FFI definition?

Do we actually need to pass these from Rust or can we just set them to default values on the C side?

Reply:
Seems like we don't need to pass these from Rust. Updated as you suggested.